File: src/main/java/orderbook/orderbook/parameters/MaterialParameter.java
package orderbook.orderbook.parameters;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.PlayerInventory;

public class MaterialParameter implements ItemParameter {

    private final Material material;

    public MaterialParameter(Material item) {

        assert item.isItem() : "Attempted to create MaterialParameter with non item Material";

        material = item;
    }

    @Override
    public ItemStack toItemStack(PlayerInventory inv, int count) {

        // if one of these is not air or zero make sure they both are
        if(material == Material.AIR || count == 0) {
            return new ItemStack(Material.AIR, 0);
        }

        return new ItemStack(material, count);
    }
}

File: src/main/java/orderbook/orderbook/parameters/MaterialParameter.kt
package orderbook.orderbook.parameters

import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.PlayerInventory

class MaterialParameter(mat: Material) : ItemParameter {
    private val material: Material

    init {
        assert(mat.isItem) { "Attempted to create MaterialParameter with non item Material" }
        material = mat
    }

    override fun toItemStack(inv: PlayerInventory, count: Int): ItemStack {
        // if one of these is not air or zero make sure they both are
        return if (material == Material.AIR || count == 0) {
            ItemStack(Material.AIR, 0)
        } else ItemStack(material, count)
    }
}